lambdas/report-user-identity/src/main/java/uk/gov/di/ipv/core/reportuseridentity/ReportUserIdentityHandler.java

```diff
@@ -101,116 +106,124 @@ public void handleRequest(InputStream inputStream, OutputStream outputStream, Co
         } finally {
             LOGGER.info(LogHelper.buildLogMessage("Writing output with result summary."));
             OBJECT_MAPPER.enable(SerializationFeature.INDENT_OUTPUT);
-            OBJECT_MAPPER.writeValue(outputStream, reportProcessingResult.build());
+            OBJECT_MAPPER.writeValue(outputStream, reportProcessingResult);
         }
     }
 
     private void scanToExtractUniqueUserIdFromTacticalStore(
             ReportProcessingRequest reportProcessingRequest,
-            ReportProcessingResult.ReportProcessingResultBuilder reportProcessingResult,
+            ReportProcessingResult reportProcessingResult,
             Context context)
             throws BatchProcessingException, StopBeforeLambdaTimeoutException {
-        if (reportProcessingRequest.continueUniqueUserScan()) {
-            LOGGER.info(
-                    LogHelper.buildLogMessage("Retrieving userIds from tactical store db table."));
+        LOGGER.info(LogHelper.buildLogMessage("Retrieving userIds from tactical store db table."));
 
-            for (var page :
-                    vcStoreItemScanDynamoDataStore.getScannedItemsPages(
-                            DynamoDbHelper.marshallToLastEvaluatedKey(
-                                    reportProcessingRequest.tacticalStoreLastEvaluatedKey()),
-                            ATTR_NAME_USER_ID)) {
+        for (var page :
+                vcStoreItemScanDynamoDataStore.scan(
+                        DynamoDbHelper.marshallToLastEvaluatedKey(
+                                reportProcessingRequest.tacticalStoreLastEvaluatedKey()),
+                        reportProcessingRequest.pageSize(),
+                        ATTR_NAME_USER_ID)) {
 
-                reportUserIdentityScanDynamoDataStore.createOrUpdate(
-                        page.items().stream()
-                                .map(VcStoreItem::getUserId)
-                                .distinct()
-                                .map(
-                                        usrId ->
-                                                new ReportUserIdentityItem(
-                                                        usrId, null, null, null, null))
-                                .toList());
+            reportUserIdentityScanDynamoDataStore.createOrUpdate(
+                    page.items().stream()
+                            .map(VcStoreItem::getUserId)
+                            .distinct()
+                            .map(usrId -> new ReportUserIdentityItem(usrId, null, null, null, null))
+                            .toList());
 
-                reportProcessingResult.tacticalStoreLastEvaluatedKey(
-                        DynamoDbHelper.unmarshallLastEvaluatedKey(page.lastEvaluatedKey()));
+            reportProcessingResult.setTacticalStoreLastEvaluatedKey(
+                    DynamoDbHelper.unmarshallLastEvaluatedKey(page.lastEvaluatedKey()));
+            reportProcessingResult.addTacticalVcsEvaluated(page.items().size());
```

Comment:
Seems this will always have page size value. Is that u wanted or want to capture items processed in that run before break.

Reply:
This is adding the page size to the total, so should be capturing the total processed items (unless it fails due to an exception).

You're right, the page size will mostly be constant - unless the total size of items exceeds 1MB or we've reached the end of the scan.